src/Resources/config/services/subscribers.xml

```diff
@@ -5,44 +5,25 @@
         https://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
         <!--Event subscribers-->
-        <service id="Adyen\Shopware\Subscriber\PaymentSubscriber">
-            <argument type="service" id="Adyen\Shopware\Provider\AdyenPluginProvider"/>
-            <argument type="service" id="Adyen\Shopware\Service\PaymentMethodsFilterService"/>
-            <argument type="service" id="Adyen\Shopware\Service\PaymentStateDataService"/>
-            <argument type="service" id="Symfony\Component\Routing\RouterInterface"/>
-            <argument type="service" id="Adyen\Shopware\Service\Repository\SalesChannelRepository"/>
-            <argument type="service" id="Adyen\Shopware\Service\ConfigurationService"/>
-            <argument type="service" id="Adyen\Shopware\Service\PaymentMethodsService"/>
-            <argument type="service" id="request_stack"/>
-            <argument type="service" id="Shopware\Core\Checkout\Cart\CartPersister"/>
-            <argument type="service" id="Shopware\Core\Checkout\Cart\CartCalculator"/>
-            <argument type="service" id="Shopware\Core\System\SalesChannel\SalesChannel\ContextSwitchRoute"/>
-            <argument type="service" id="Shopware\Core\System\SalesChannel\Context\SalesChannelContextFactory"/>
-            <argument type="service" id="Adyen\Util\Currency"/>
+        <service id="Adyen\Shopware\Subscriber\PaymentSubscriber" autowire="true">
+            <argument key="$router" type="service" id="Symfony\Component\Routing\RouterInterface"/>
+            <argument key="$requestStack" type="service" id="request_stack"/>
+            <argument key="$cartPersister" type="service" id="Shopware\Core\Checkout\Cart\CartPersister"/>
+            <argument key="$contextSwitchRoute" type="service" id="Shopware\Core\System\SalesChannel\SalesChannel\ContextSwitchRoute"/>
```

Comment:
Why no autowire these too?

Reply:
Hi @peterojo, 

Thank you for the review. You can find the explanation below for each class injection.

`ContextSubscriber` implements `EventSubscriberInterface` which expects `AbstractContextSwitchRoute` on the constructor, which can't be auto wirable since it's an abstract class. It's required to inject `ContextSwitchRoute` manually. Similarly, that interface expects `AbstractCartPersister` which requires `CartPersister` to be injected manually.

Therefore, we can't auto wire those dependencies on the given classes.